File: generators/app/templates/java/app/src/main/java/org/onepointzero/androidstarterjava/application/GetStarterData.java
package <%= packageName %>.application;

import <%= packageName %>.domain.StarterData;
import <%= packageName %>.domain.StarterDataRepository;

import io.reactivex.Single;

public class GetStarterData {
    private StarterDataRepository starterDataRepository;

    public GetStarterData(StarterDataRepository starterDataRepository) {
        this.starterDataRepository = starterDataRepository;
    }

    public Single<String> execute() {
        return starterDataRepository
                .getStarterData()
                .map(StarterData::getMessage);
    }
}

File: generators/app/templates/kotlin/app/src/main/java/org/onepointzero/androidstarterkotlin/application/GetStarterData.kt
package <%= packageName %>.application

import io.reactivex.Single
import <%= packageName %>.domain.StarterData
import <%= packageName %>.domain.StarterDataRepository

class GetStarterData(private val starterDataRepository: StarterDataRepository) {

    fun execute(): Single<String> {
        return starterDataRepository
                .get()
                .map(StarterData::message)
    }
}
